plugins/janus_videoroom.c

```diff
@@ -987,11 +1010,15 @@ void janus_videoroom_destroy_session(janus_plugin_session *handle, int *error) {
 		if(session->participant_type == janus_videoroom_p_type_publisher) {
 			/* Get rid of publisher */
 			janus_videoroom_publisher *p = (janus_videoroom_publisher *)session->participant;
+			janus_mutex_lock(&session->mutex);
 			session->participant = NULL;
+			janus_mutex_unlock(&session->mutex);
+			janus_videoroom_leave_or_unpublish(p, TRUE);
 			if(p->room) {
+				/* Don't clear p->room.  Another thread calls janus_videoroom_leave_or_unpublish,
+					 too, and there is no mutex to protect this change. */
 				janus_refcount_decrease(&p->room->ref);
```

Comment:
I call UnpublishOwnFeed() and .destroy() on each session onBeforeUnload so perhaps that causes it sometimes. It's not something I was able to consistently replicate yet though so it might occasionally pop up when sending 'detach' or 'unpublish' gets delayed.
this change fixed that indeed though so we don't have to trust the client to not crash the gateway 👍